ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/WorkloadTableToolbar.tsx

```diff
@@ -55,19 +64,22 @@ function WorkloadTableToolbar({
         const newResourceFilter = searchFilter[category] as string[];
         const chipKey = typeof chip === 'string' ? chip : chip.key;
         newSearchFilter[category] = newResourceFilter.filter((fil: string) => fil !== chipKey);
-        setSearchFilter(newSearchFilter);
+        onChangeSearchFilter(newSearchFilter);
     }
 
     function onDeleteGroup(category: FilterType | Resource) {
         const newSearchFilter = { ...searchFilter };
         delete newSearchFilter[category];
-        setSearchFilter(newSearchFilter);
+        onChangeSearchFilter(newSearchFilter);
     }
 
     function onDeleteAll() {
-        setSearchFilter({});
+        onChangeSearchFilter({});
     }
 
+    // The `onChangeSearchFilter` function is intentionally not used in place of `setSearchFilter` below since
```

Comment:
I figured it was worth calling out that this is explicitly not used in the effect since it might look like an accidental omission. Resetting the current page number is handled via user action in the DefaultFilterModal's parent component.